extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1002,20 +1008,43 @@ public String getHost() {
 
     @Nonnull
     @Override
-    public String getPrivacy() {
-        return "";
+    public Privacy getPrivacy() {
+        boolean isUnlisted = playerResponse
+                .getObject("microformat")
+                .getObject("playerMicroformatRenderer")
+                .getBoolean("isUnlisted");
+        return isUnlisted ? Privacy.UNLISTED : Privacy.PUBLIC;
     }
 
     @Nonnull
     @Override
     public String getCategory() {
-        return "";
+        try {
+            return playerResponse.getObject("microformat")
+                    .getObject("playerMicroformatRenderer")
+                    .getString("category");
+        } catch (Exception ignored) {
+            return "";
+        }
     }
 
     @Nonnull
     @Override
     public String getLicence() {
-        return "";
+        try {
+            final JsonObject metadataRowRenderer = getVideoSecondaryInfoRenderer()
+                    .getObject("metadataRowContainer").getObject("metadataRowContainerRenderer").getArray("rows")
+                    .getObject(0)
+                    .getObject("metadataRowRenderer");
+
+            final String probablyLicence = getTextFromObject(metadataRowRenderer
+                    .getArray("contents").getObject(0));
+
+            return probablyLicence == null || !probablyLicence.contains("Creative Commons")
+                    ? "YouTube licence" : probablyLicence;
```

Comment:
YouTube themselves call it "YouTube license" so I wouldn't change the name even though you prefer "licence"

Reply:
No, look at here:
They also call CC `Creative Commons Attribution licence (reuse allowed)`
they also use `licence` in the Terms of Service

https://web.archive.org/web/20190723193439/https://www.youtube.com/static?template=terms&hl=en
> Rights you licen**c**e
> 8.1 When you upload or post Content to YouTube, you grant:
> - to YouTube, a worldwide, non-exclusive, royalty-free, transferable licen**c**e (with right to sub-licence) to use, reproduce, distribute, prepare derivative works of, display, and perform that Content in connection with the provision of the Service and otherwise in connection with the provision of the Service and YouTube's business, including without limitation for promoting and redistributing part or all of the Service (and derivative works thereof) in any media formats and through any media channels;
> - to each user of the Service, a worldwide, non-exclusive, royalty-free licen**c**e to access your Content through the Service, and to use, reproduce, distribute, prepare derivative works of, display and perform such Content to the extent permitted by the functionality of the Service and under these Terms.